src/plugins/topic_viewer/TopicViewer.hh

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <QStandardItem>
+#include <QStandardItemModel>
+#include <QHash>
+#include <QByteArray>
+#include <QString>
+#include <QModelIndex>
+#include <string>
+#include <vector>
+#include <deque>
+#include <memory>
+
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/Plugin.hh>
+
+#include <ignition/msgs.hh>
+#include <ignition/transport.hh>
```

Comment:
It's preferable to never use these catch-all headers because they bring a lot of things that we won't use. Instead, we should just include the headers we need.

For example: `#include <ignition/msgs/Factory.hh>`

So let's remove all of these:

```
#include <ignition/msgs.hh>
#include <ignition/transport.hh>
#include <ignition/common.hh>
#include <ignition/gui.hh>
```